### DISCUSSION TITLE
qwerty learner 桌面版

### DISCUSSION THREAD

**Discussion by tangshimin** [RealKai42/qwerty-learner#197] [Ideas]
我基于 qwerty-learner 写了一个桌面版，增加了如下新功能：

1. 可以用[电影生成词库](https://github.com/tangshimin/typing-learner/wiki/%E4%BB%8E-MKV-%E8%A7%86%E9%A2%91%E7%94%9F%E6%88%90%E8%AF%8D%E5%BA%93)或[字幕生成词库](https://github.com/tangshimin/typing-learner/wiki/%E4%BB%8E%E6%96%87%E6%A1%A3%E7%94%9F%E6%88%90%E8%AF%8D%E5%BA%93)，让每个单词都有具体的语境。有了这个功能，今后就可以一边追美剧一边学英语了。
  
2. 可以用[英文文档生成词库](https://github.com/tangshimin/typing-learner/wiki/%E4%BB%8E%E6%96%87%E6%A1%A3%E7%94%9F%E6%88%90%E8%AF%8D%E5%BA%93)，比如你需要学习 TypeScript 最新的文档又没有中文版，你就可以用英文文档生成词库，然后过滤掉一些你熟悉的词，剩下的就是一些陌生的单词。把陌生单词学习几遍之后再阅读技术文档就更流畅了。
  
3. 基于文档的词库可以链接用 MKV 视频或字幕生成的词库。
  
4. 可以删除和修改词库。
  
5. 音频资源会自动缓存。
  
6. 学习完一章之后，可以选择进入默写模式，在默写模式整个章节的单词是重新随机排序的，默写完了会出现默写结果。
  
7. 学习完整个词库之后，还有一个随机排序整个词库的功能。
  
8. 单词可以重复输入多次。
  
9. 字幕和单词英语定义也可以重复输入多次。
  
10. 增加了词性，词形和英语定义等属性，可以自由的关闭和隐藏。

11. 默认使用 Enter 键切换下一个单词，如果要使用类似于网页版的自动切换，可以使用 Ctrl + A 开启自动切换。

#### [下载地址](https://github.com/tangshimin/typing-learner/releases)

https://user-images.githubusercontent.com/16540656/162030268-dddfa054-3fe1-414f-8563-73ca9f841d89.mp4

**Comment by tangshimin** [comment]
@Kaiyiwing

**Comment by RealKai42** [comment]
酷呀， 功能上非常丰富了，UI 上可以再做一些美化
这个是用什么技术栈实现的？有计划开源么，开源的话我可以挂到 issue 区置顶～

**Comment by tangshimin** [comment]
代码是用 kotlin + compose 写的已经上传到 Github 现在是 私有仓库，我可以使用 qwerty-leaner-desktop 作为项目名称吗? 可以的话我会尽快将代码遵循 GPL-3 协议公开。

**Reply by RealKai42** [reply]
当然可以

**Reply by tangshimin** [reply]
可以使用 qwerty-leaner 作为桌面应用程序的名称吗？

**Comment by tangshimin** [comment]
现在还有一些 bug

**Comment by tangshimin** [comment]
@Kaiyiwing  已发现的 bug 都修复了。 我给桌面版增加了一个音效，用于完成当前章节后播放，这个音效是 CC-BY 保留署名许可协议。键盘击键音效是从哪里下载的? 也是 CC-BY 保留署名许可协议吗?

**Comment by RealKai42** [comment]
声音是由 @chengluyu  添加的，我不太确定相关开源协议

**Comment by tangshimin** [comment]
@Kaiyiwing [代码库](https://github.com/tangshimin/qwerty-learner-desktop) 公开了，windows 版 [下载地址](https://github.com/tangshimin/qwerty-learner-desktop/releases/tag/v1.0.0-beta)

**Comment by tangshimin** [comment]
更新了一个版本，https://github.com/tangshimin/qwerty-learner-desktop/tags

**Comment by RealKai42** [comment]
等稳定后可以提一个 issues 在这个项目，我可以帮忙置顶宣传一下

**Reply by tangshimin** [reply]
非常感谢！

**Comment by sdu-gyf** [comment]
好酷！
